payu/fsops.py

```diff
@@ -62,6 +63,27 @@ def movetree(src, dst, symlinks=False):
     shutil.rmtree(src)
 
 
+class DuplicateKeyWarnLoader(yaml.SafeLoader):
+    def construct_mapping(self, node, deep=False):
+        """Add warning for duplicate keys in yaml file, as currently
+        PyYAML overwrites duplicate keys even though in YAML, keys
+        are meant to be unique
+        """
+        mapping = {}
+        for key_node, value_node in node.value:
+            key = self.construct_object(key_node, deep=deep)
+            value = self.construct_object(value_node, deep=deep)
+            if key in mapping:
+                warnings.warn(
+                    "Duplicate key found in config.yaml: "
+                    f"key '{key}' with value '{value}' "
+                    f"(original value: '{mapping[key]}')"
+                )
```

Comment:
Do you need to specify that `mapping[key]` will overwrite `value` in this warning?

Reply:
Yeah, in PyYAML the subsequent `value` overwrites the `mapping[key]` value. So I've updated the warning message in cddbef6. 

The annoying thing is that `ruamel.yaml` removes the subsequent duplicate key-value. The hope is the warnings from reading `config.yaml` in `fsops.py` which runs at every `payu` command would help prevent the duplicate keys existing in the first place..